pkg/openstack/nova.go

```diff
@@ -45,9 +51,110 @@ func ReconcileNova(ctx context.Context, instance *corev1beta1.OpenStackControlPl
 			return res, err
 		}
 		instance.Status.Conditions.Remove(corev1beta1.OpenStackControlPlaneNovaReadyCondition)
+		instance.Status.Conditions.Remove(corev1beta1.OpenStackControlPlaneExposeNovaReadyCondition)
 		return ctrl.Result{}, nil
 	}
 
+	// add selector to service overrides
+	for _, endpointType := range []service.Endpoint{service.EndpointPublic, service.EndpointInternal} {
+		// NovaAPI
+		if instance.Spec.Nova.Template.APIServiceTemplate.Override.Service == nil {
+			instance.Spec.Nova.Template.APIServiceTemplate.Override.Service = map[string]service.RoutedOverrideSpec{}
+		}
+		instance.Spec.Nova.Template.APIServiceTemplate.Override.Service[string(endpointType)] =
+			AddServiceComponentLabel(
+				ptr.To(instance.Spec.Nova.Template.APIServiceTemplate.Override.Service[string(endpointType)]),
+				nova.Name+"-api")
+
+		// cell NoVNCProxy service override
+		for cellName := range instance.Spec.Nova.Template.CellTemplates {
+			cellTemplate := instance.Spec.Nova.Template.CellTemplates[cellName]
+			if cellTemplate.NoVNCProxyServiceTemplate.Override.Service == nil {
```

Comment:
to reduce clutter in the generate OpenstackControlPlane CR you can skip adding override for all the cells where novncproxy is disabled.
```golang
			if cellTemplate.NoVNCProxyServiceTemplate.Enabled == False {
                continue
            }
```

Reply:
done